src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -141,6 +141,16 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
         site_config.app_settings.append(NameValuePair(name="WEBSITE_NODE_DEFAULT_VERSION",
                                                       value=node_default_version))
 
+    # This is to keep the existing appsettings for a newly created webapp on existing webapp name.
+    name_validation = client.check_name_availability(name, 'Site')
```

Comment:
why not directly call get? If get return with none, means no existing site, else appending to existing app settings. this saves a rest call.

Reply:
@yungezz - agreed  that making a `get` call will work but this is an expensive call where it returns an entire webapp object and we need to hold it through out. But in my scenario I will make second call only when the app already exists, and the second call is only to fetch the appsettings. And it will not occupy more memory even though we hold it. Both the rest calls are light weight here.